scripts/Calc_hru_wgtavg_nc.py

```diff
@@ -234,8 +172,21 @@ def usage():
       varname = args[2]  
       nc_out  = args[3]
 
-      wgtVal=compAvgVal(nc_wgt,nc_in,varname)
-      writeNetCDFData(nc_out, wgtVal, varname)     
+      func=partial(compAvgVal, nc_wgt, nc_in, varname)
+
+      pool = mp.Pool(processes=NPROC,maxtasksperchild=1)
+      results=pool.map(func,range(NPROC))
+      result=np.column_stack(results)
+
+      with xr.open_dataset(nc_in) as ds:
+        timedata  = ds.coords[TDIM]     # Get time coordinate variable
+      with xr.open_dataset(nc_wgt) as ds:
+        hrudata = ds[HDIM].values
+
+      encoding={varname: {'dtype': 'float32', '_FillValue': -9999.0}}
+      foo = xr.DataArray(result, coords=[('time',timedata), ('hru',hrudata)], name=varname)
+      foo.encoding = encoding[varname]
+      foo.to_netcdf('test.nc')
```

Comment:
just a reminder to provide a way to programmatically determine the output file name.

Reply:
replaced 'test.nc' with nc_out